src/components/MobileApp/ModelFilterForm.js

```diff
@@ -0,0 +1,254 @@
+                All inventory pools
+              </option>
+              {pools.map((p) => (
+                <option value={p.id} key={p.id}>
+                  {p.name}
+                </option>
+              ))}
+            </select>
+          </div>
+        </label>
+        <div className="form-group">
+          <div className="row">
+            <span className="text-xs col-3 col-form-label">Time Span</span>
+            <div className="col-9">
+              <label className="custom-control custom-checkbox">
+                <input
+                  type="checkbox"
+                  name="only-available"
+                  checked={availableBetween}
+                  onChange={(e) => setAvailableBetween((prevBool) => !prevBool)}
+                  className="custom-control-input"
+                />
+                <span className="custom-control-label">Show available only</span>
+              </label>
+            </div>
+          </div>
+        </div>
+        {availableBetween && (
+          <div>
+            <div className="form-group">
+              <label className="row">
+                <span className="text-xs col-3 col-form-label">From </span>
+                <div className="col-9">
+                  <input
+                    placeholder="From"
+                    name="start-date"
+                    type="date"
+                    className="form-control "
+                    required
+                    defaultValue={startDate}
+                    onChange={(e) => setStartDate(e.target.value)}
+                  />
+                </div>
+              </label>
+            </div>
+            <div className="form-group">
+              <label className="row">
+                <span className="text-xs col-3 col-form-label">Until </span>
+                <div className="col-9">
+                  <input
+                    placeholder="Until"
+                    name="end-date"
+                    type="date"
+                    className="form-control "
+                    required
+                    defaultValue={endDate}
+                    onChange={(e) => setEndDate(e.target.value)}
+                  />
+                </div>
+              </label>
+            </div>
+            <div className="form-group">
+              <label className="row">
+                <span className="text-xs col-3 col-form-label">Quantity </span>
+                <div className="col-9">
+                  <input
+                    type="number"
+                    min={1}
+                    name="quantity"
+                    placeholder="Quantity"
+                    className="form-control "
+                    defaultValue={quantity}
+                    onChange={(e) => setQuantity(e.target.value)}
+                  />
+                </div>
+              </label>
+            </div>
+          </div>
+        )}
+        <button
+          type="submit"
+          onClick={(e) => {
```

Comment:
instead of `onClick` on the submit button, this should move to `onSubmit` on the `<form>`.